src/blocks.zig

```diff
@@ -409,29 +409,41 @@ pub const TickFunctions = struct {
 };
 
 pub const TickEvent = struct {
-	function: ?*const TickFunction,
+	function: *const TickFunction,
 	chance: f32,
 	tryEvery: u16,
 
 	pub fn loadFromZon(_allocator: main.heap.NeverFailingAllocator, zon: ZonElement) []TickEvent {
-		const tickEvents = _allocator.alloc(TickEvent, zon.toSlice().len);
+		var tickEvents = _allocator.alloc(TickEvent, zon.toSlice().len);
 
-		for(zon.toSlice(), 0..) |tickEventZon, i| {
+		var n: usize = 0;
```

Comment:
Please just use a `List` instead of emulating it.
Then you can just use `toOwnedSlice` in the end.